src/components/ItemDrawer.tsx

```diff
@@ -0,0 +1,175 @@
+        await DataStore.save(rows[i]);
+        console.log("Item sent successfully!");
+      } catch (error) {
+        console.log("Error sending row", error);
+      }
+    }
+    setRows(() => []);
+  }
+
+  return (
+    <>
+      <button
+        className="add-button2"
+        type="button"
+        onClick={() => setIsDrawerOpen(true)}
+      >
+        <div className="button-layout2">
+          {/* <div className="plus" /> */}
+          <AddIcon sx={{ color: "white", fontSize: 12 }} />
+          <div className="add-text2">Add Items</div>
+        </div>
+      </button>
+      <Drawer
+        anchor="bottom"
+        hideBackdrop
+        open={isDrawerOpen}
+        onClose={() => {
+          setIsDrawerOpen(false);
+          setRows(() => []);
+        }}
+      >
+        <div>
+          <button
+            className="back-button"
+            type="button"
+            onClick={() => {
+              setIsDrawerOpen(false);
+              setRows(() => []);
+            }}
+            aria-label="back"
+          />
+          <div className="summary-row">
+            <div className="summary-text">
+              Donation Summary: {rows.length} Items
+            </div>
+            <button
+              className="add-button"
+              type="button"
+              onClick={() => setRows([...rows, testRow])}
+              // need to change this to get new item from Add Item modal
+            >
+              <div className="button-layout">
+                {/* <div className="plus" /> */}
+                <AddIcon sx={{ color: "white", fontSize: 12 }} />
+                <div className="add-text">Add Item</div>
+              </div>
+            </button>
+          </div>
+          <div className="data-table">
+            <DataGrid
+              rows={rows}
+              columns={columns}
+              getRowHeight={() => "auto"}
+              paginationModel={paginationModel}
+              onPaginationModelChange={setPaginationModel}
+              pageSizeOptions={[10, 25, 50]}
+              sx={{
+                "&.MuiDataGrid-root--densityCompact .MuiDataGrid-cell": {
+                  py: "8px",
+                },
+                "&.MuiDataGrid-root--densityStandard .MuiDataGrid-cell": {
+                  py: "15px",
+                },
+                "&.MuiDataGrid-root--densityComfortable .MuiDataGrid-cell": {
+                  py: "22px",
+                },
+              }}
+            />
+          </div>
+          <div className="end-buttons">
+            <button
+              className="cancel-button"
+              type="button"
+              onClick={() => {
+                setIsDrawerOpen(false);
+                setRows(() => []);
+              }}
+            >
+              Cancel
+            </button>
+            <button
+              className="send-button"
+              type="button"
+              disabled={isDisabled}
```

Comment:
you can have a conditional class where if disabled=true, the button is "grayed" out. This is just a styling preference though

Reply:
The buttons in general are all also pretty bare bones, I unset all the default button styling so they don't press down or change color or anything when clicked. Partly because I wasn't sure if we want to have some unified button styling for the entire site?

Reply:
We could and we should do that. You could use MUI's button to help with this. Honestly should've used mui since the start. We will probably have to refractor all the buttons to mui for consistency.